### DISCUSSION TITLE
Limit results returned by tag value autocompletion queries to InfluxDB

### DISCUSSION THREAD

**Discussion by sciurus** [grafana/grafana#38325] [Data Sources]
When you are graphically building a query for InfluxDB and choose a tag for a WHERE clause, Grafana issues a `SHOW TAG VALUES` query in order to populate the autocompletion list. If there are a large number of tag values, this query can take a long time and even cause InfluxDB to run out of memory and be killed. See #4441 for a report of this.

InfluxDB supports a [LIMIT](https://docs.influxdata.com/influxdb/v1.8/query_language/explore-schema/#show-tag-values) clause that should avoid this problem. For example, instead of issuing `SHOW TAG VALUES WITH KEY = "host"` Grafana could issue `SHOW TAG VALUES WITH KEY = "host" LIMIT 1000`. 

1000 may not be the right value, but probably there is a reasonable limit beyond which Grafana itself would perform poorly.

**Comment by Elfo404** [comment]
@ryantxu I think this might be of your interest

**Comment by Esity** [comment]
This could potentially be added to the datasources page as a config item? I know that with a bunch of tag values, this can cause performance issues however we did request that InfluxDB add a meta timeout to InfluxDB which was completed for the 1.8.x line

https://github.com/influxdata/influxdb/issues/20737

https://docs.influxdata.com/influxdb/v1.8/about_the_project/releasenotes-changelog/#bug-fixes

> Update meta queries (for example, SHOW TAG VALUES, SHOW TAG KEYS, SHOW SERIES CARDINALITY, SHOW MEASUREMENT CARDINALITY, and SHOW MEASUREMENTS) to check the query context when possible to respect timeout values set in the query-timeout configuration parameter. Note, meta queries will check the context less frequently than regular queries, which use iterators, because meta queries return data in batches.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
